editor/src/dev/yourkit.clj

```diff
@@ -141,42 +222,69 @@
 ;; Allocation recording
 ;; -----------------------------------------------------------------------------
 
-(def default-allocation-recording-settings (AllocationRecordingSettings.))
+(def ^AllocationProfilingSettings default-allocation-profiling-settings (AllocationProfilingSettings.))
 
-(definline start-allocation-recording!
+(defmacro start-allocation-recording!
   "Starts recording memory allocations. The call clears previously recorded
   allocation data."
-  [^Controller controller]
-  `(.startAllocationRecording ~controller default-allocation-recording-settings))
+  ([] (start-allocation-recording! (controller)))
+  ([^Controller controller]
+   `(.startAllocationProfiling ~(with-meta controller {:tag `Controller}) default-allocation-profiling-settings)))
 
-(definline stop-allocation-recording!
+(defmacro stop-allocation-recording!
   "Stops recording memory allocations. Has no effect unless allocation recording
   is running."
-  [^Controller controller]
-  `(.stopAllocationRecording ~controller))
+  ([] (stop-allocation-recording! (controller)))
+  ([^Controller controller]
+   `(.stopAllocationProfiling ~(with-meta controller {:tag `Controller}))))
 
-(definline capture-allocation-snapshot!
+(defmacro capture-allocation-snapshot!
   "Writes the profiling results to a snapshot file on disk. Includes both any
   recorded CPU profiling results, and recorded memory allocations. Returns the
   path to the snapshot file."
-  ^String [^Controller controller]
-  `(let [snapshot-path# (.captureMemorySnapshot ~controller)]
-     (swap! snapshot-paths-atom conj snapshot-path#)
-     snapshot-path#))
+  (^String [] (capture-allocation-snapshot! (controller)))
+  (^String [^Controller controller]
+   `(let [snapshot-path# (.captureMemorySnapshot ~(with-meta controller {:tag `Controller}))]
+      (push-snapshot-path! snapshot-path#))))
+
+(defmacro capture-garbage-collected-allocation-snapshot!
+  "Forces a garbage collection, then writes the profiling results to a snapshot
+  file on disk. Includes both any recorded CPU profiling results, and recorded
+  memory allocations. Returns the path to the snapshot file."
+  (^String [] (capture-garbage-collected-allocation-snapshot! (controller)))
+  (^String [^Controller controller]
+   `(let [^Controller controller# ~controller]
+      (force-garbage-collection! controller#)
+      (force-garbage-collection! controller#)
+      (capture-allocation-snapshot! controller#))))
```

Comment:
It's intended to call `force-garbage-collection!` twice here. Seems like the first call will sometimes not take care of it all.